### PR TITLE
Feature/apm integration

### PR DESCRIPTION
**Pull request by SylvainJuge**
- single `elasticsearch` service for the whole cluster
- map a few tasks attributes on the OTel span
- hack the span/transaction type to use `elasticsearch` as span/transaction type
- hack the span destination to create one destination per node in the map (will also be included in dependencies)

### CODE REVIEW

**Comment by DaveCTurner** [comment]
@elasticmachine ok to test

**Review by AthenaEryma — COMMENTED** [review]
LGTM, I was thinking we should add something like this. I only didn't +1 this because I think we can check this in `ApmIT`, but I'm not going to -1 this over that either because it's On Week. But we should at least add a TODO about it?

**Comment by SylvainJuge** [comment]
@elasticsearchmachine run elasticsearch-ci/part-2

File: x-pack/plugin/apm-integration/src/main/java/org/elasticsearch/xpack/apm/APMTracer.java [hunk 1]
```diff
@@ -214,6 +219,21 @@ public void onTraceStarted(Traceable traceable) {
                     );
                 }
             }
+
+            // hack transactions to avoid the 'custom' type
+            spanBuilder.setAttribute("type", "elasticsearch");
+
+            // hack spans to avoid the 'app' span.type, will make it use external/elasticsearch
+            // also allows to set destination resource name in map
+            spanBuilder.setAttribute("messaging.system", "elasticsearch");
```
Comment (tlrx) [inline x-pack/plugin/apm-integration/src/main/java/org/elasticsearch/xpack/apm/APMTracer.java:228]: should we use SemanticsAttributes here?

**Review by tlrx — COMMENTED** [review]
Looks good, can we add a test (or update the existing ones)?

File: x-pack/plugin/apm-integration/src/main/java/org/elasticsearch/xpack/apm/APMTracer.java (hunk 1 above)
Comment (SylvainJuge) [reply x-pack/plugin/apm-integration/src/main/java/org/elasticsearch/xpack/apm/APMTracer.java:228]: Yes good point, just fixed that.

**Comment by SylvainJuge** [comment]
:man_facepalming: just merged it by accident, but it's only cosmetics so should be fine.

**Pull request merged**